lang/csharp/src/apache/main/AvroDecimal.cs

```diff
@@ -0,0 +1,983 @@
+            var scale = bytes[14];
+
+            if (bytes[15] == 128)
+                unscaledValue *= BigInteger.MinusOne;
+
+            UnscaledValue = unscaledValue;
+            Scale = scale;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given int.
+        /// </summary>
+        /// <param name="value">The int value.</param>
+        public AvroDecimal(int value)
+            : this(new BigInteger(value), 0) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given long.
+        /// </summary>
+        /// <param name="value">The long value.</param>
+        public AvroDecimal(long value)
+            : this(new BigInteger(value), 0) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given unsigned int.
+        /// </summary>
+        /// <param name="value">The unsigned int value.</param>
+        public AvroDecimal(uint value)
+            : this(new BigInteger(value), 0) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given unsigned long.
+        /// </summary>
+        /// <param name="value">The unsigned long value.</param>
+        public AvroDecimal(ulong value)
+            : this(new BigInteger(value), 0) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given <see cref="BigInteger"/>
+        /// and a scale.
+        /// </summary>
+        /// <param name="unscaledValue">The double value.</param>
+        /// <param name="scale">The scale.</param>
+        public AvroDecimal(BigInteger unscaledValue, int scale)
+        {
+            UnscaledValue = unscaledValue;
+            Scale = scale;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given array of bytes.
+        /// </summary>
+        /// <param name="value">The byte array.</param>
+        public AvroDecimal(byte[] value)
+        {
+            byte[] number = new byte[value.Length - 4];
+            byte[] flags = new byte[4];
+
+            Array.Copy(value, 0, number, 0, number.Length);
+            Array.Copy(value, value.Length - 4, flags, 0, 4);
+
+            UnscaledValue = new BigInteger(number);
+            Scale = BitConverter.ToInt32(flags, 0);
+        }
+
+        /// <summary>
+        /// Gets the unscaled integer value represented by the current <see cref="AvroDecimal"/>.
+        /// </summary>
+        public BigInteger UnscaledValue { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the current <see cref="AvroDecimal"/> represents an even number.
+        /// </summary>
+        public bool IsEven
+        {
+            get { return UnscaledValue.IsEven; }
```

Comment:
this is not true if scale is != 0 right? same comment with other properties below.

Reply:
Ah, yeah, of course. Decimals don't have the notion of even and odd. As per your previous comment, I'll remove these properties rather than test on the `Scale` which may be confusing.

Reply:
I have removed these properties.